test/sharness/t0080-repo.sh

```diff
@@ -0,0 +1,129 @@
+#!/bin/sh
+#
+# Copyright (c) 2014 Jeromy Johnson
+# MIT Licensed; see the LICENSE file in this repository.
+#
+
+test_description="Test ipfs repo operations"
+
+. lib/test-lib.sh
+
+test_init_ipfs
+test_launch_ipfs_daemon
+
+test_expect_success "'ipfs add afile' succeeds" '
+	echo "some text" > afile
+	HASH=`ipfs add -q afile`
+	echo -n $HASH > hashfile
+'
+
+test_expect_success "added file was pinned" '
+	ipfs pin ls -type=recursive | grep $HASH
+'
+
+test_expect_success "'ipfs repo gc' doesnt remove file" '
+	echo -n "" > empty
+	ipfs repo gc > gc_out_actual
+	test_cmp empty gc_out_actual
+	ipfs cat $HASH > out
+	test_cmp out afile
+'
```

Comment:
We're also mixing `||` in, so at least `&&` is more explicit:
```
test_expect_success "'ipfs add --help' output looks good" '
	egrep "ipfs add.*<path>" actual >/dev/null ||
	fsh cat actual
'
```

if `&&` is only there for the reader, then we can remove it, but let's confirm. @chriscool thoughts? is the `&&` for the reader or is there a case we're not noticing here?


(making sure our tests are perfectly solid is very important, if they start behaving slightly differently in different platforms, it will cause a lot of pain)